### ISSUE TITLE
Not playing on iOS after upgrade

### ISSUE THREAD

**Issue by maendamedia** [DoubleSymmetry/react-native-track-player#637]
After upgrading the module to 1.1.4, iOS stopped playing. Android is fine.

I also installed the React-native-swift package.

Is there something i miss here?

edit: it seems 1.1.0 does work, all above that version do not

**Comment by maendamedia** [comment]
Any ideas?

**Comment by moonstruck** [comment]
For me everything is working fine after upgrade. Only breaking change is `playback-state` event no longer firing in iOS.

**Comment by maendamedia** [comment]
Well i needed to rebuild most of the app to get it working. Guess i did not make it as it supposed to be :)

Only problem is still that i need to press play twice to get music playing. Any idea's how to fix that?

```
async function togglePlayback() {

    const currentTrack = await TrackPlayer.getCurrentTrack();
    if (currentTrack == null) {
      await TrackPlayer.reset();
      await TrackPlayer.add({
        id: 'stationId',
          url: playerUrl,
          title: 'Loading...',
          artist: 'Loading...',
        artwork: require('../assets/track_image.png')
      });
      await TrackPlayer.play();
      setIsPlaying(true)

     if (Platform.OS === 'android') {
        TrackPlayer.addEventListener('playback-metadata-received', async (e) => {
            const currentTrack = await TrackPlayer.getCurrentTrack();
            TrackPlayer.updateMetadataForTrack(currentTrack, { title: e.title, artist: e.artist });
        });
    } else {
        setInterval( () => {
            fetch(nowPlayingUrl)
                .then(r => r.json())
                .then(async (data) => {
                const currentTrack = await TrackPlayer.getCurrentTrack();
                TrackPlayer.updateMetadataForTrack(currentTrack, 
                    { 
                        title: data.nowplaying[0].title, 
                        artist: data.nowplaying[0].artist });
            });
        }, 1000 * 10); 
    }
    } else {
      if (playbackState === TrackPlayer.STATE_PAUSED) {
        await TrackPlayer.play();
        setIsPlaying(true)
      } else {
        await TrackPlayer.pause();
        TrackPlayer.reset();
        setIsPlaying(false)
      }
    }
  }
```

**Comment by maendamedia** [comment]
Anyone else has this issue?

**Comment by maendamedia** [comment]
:( can someone help me please?

**Comment by maendamedia** [comment]
Really? No one?

**Comment by dcvz** [comment]
@maendamedia it's a bit hard to follow what's going on. 1.1.4 seems to be working well for everyone. Could you make a demo repo which I can checkout and test?

**Comment by alburdette619** [comment]
@maendamedia 

> Only problem is still that i need to press play twice to get music playing.

I just solved this.  There is an undocumented config prop set with `setupPlayer` called `waitForBuffer`.  Set that to true, and it will in turn set `automaticallyWaitsToMinimizeStalling` on the AVPlayer to true (defaulting to false).  This seems to be needed for streaming hls and the like.
